// @smartstock/core-libs/esm2015/components/bottom-bar.component.js
import { Component } from '@angular/core';
export class BottomBarComponent {
    constructor() {
    }
    ngOnInit() {
    }
    activeNav(route) {
        const url = new URL(location.href);
        return url.pathname.startsWith('/' + route);
    }
}
BottomBarComponent.decorators = [
    { type: Component, args: [{
                selector: 'smartstock-bottom-bar',
                template: `
    <mat-toolbar class="fixed-bottom bottom-smartstock-bar">

      <div class="flex-grow-1 d-flex justify-content-center align-items-center flex-column">
        <button mat-icon-button *ngIf="activeNav('dashboard')" mat-flat-button routerLink="/dashboard">
          <mat-icon color="{{activeNav('dashboard')?'primary':''}}">dashboard</mat-icon>
        </button>
        <button *ngIf="!activeNav('dashboard')" mat-icon-button routerLink="/dashboard">
          <mat-icon>dashboard</mat-icon>
        </button>
        <!--    <span *ngIf="activeNav('dashboard')">Dashboard</span>-->
      </div>

      <div class="flex-grow-1 d-flex justify-content-center align-items-center flex-column">
        <button mat-icon-button *ngIf="activeNav('sale')" mat-flat-button routerLink="/sale">
          <mat-icon color="{{activeNav('sale')?'primary':''}}">shop</mat-icon>
        </button>
        <button *ngIf="!activeNav('sale')" mat-icon-button routerLink="/sale">
          <mat-icon>shop</mat-icon>
        </button>
        <!--    <span *ngIf="activeNav('sale')">Sale</span>-->
      </div>

      <div class="flex-grow-1 d-flex justify-content-center align-items-center flex-column">
        <button mat-icon-button *ngIf="activeNav('purchase')" mat-flat-button routerLink="/purchase">
          <mat-icon color="{{activeNav('purchase')?'primary':''}}">receipt</mat-icon>
        </button>
        <button *ngIf="!activeNav('purchase')" mat-icon-button routerLink="/purchase">
          <mat-icon>receipt</mat-icon>
        </button>
        <!--    <span *ngIf="activeNav('purchase')">Purchase</span>-->
      </div>

      <div class="flex-grow-1 d-flex justify-content-center align-items-center flex-column">
        <button mat-icon-button *ngIf="activeNav('stock')" mat-flat-button routerLink="/stock">
          <mat-icon color="{{activeNav('stock')?'primary':''}}">store</mat-icon>
        </button>
        <button *ngIf="!activeNav('stock')" mat-icon-button routerLink="/stock">
          <mat-icon>store</mat-icon>
        </button>
        <!--    <span *ngIf="activeNav('stock')">StockModel</span>-->
      </div>

      <div class="flex-grow-1 d-flex justify-content-center align-items-center flex-column">
        <button mat-icon-button *ngIf="activeNav('settings')" mat-flat-button routerLink="/account">
          <mat-icon color="{{activeNav('settings')?'primary':''}}">settings</mat-icon>
        </button>
        <button *ngIf="!activeNav('settings')" mat-icon-button routerLink="/account">
          <mat-icon>settings</mat-icon>
        </button>
        <!--    <span *ngIf="activeNav('settings')">Settings</span>-->
      </div>

    </mat-toolbar>
  `,
                styles: ["span{font-size:11px;font-weight:lighter}.bottom-smartstock-bar{background:#fff;box-shadow:0 -1px 7px rgba(0,0,0,.35)}"]
            },] }
];
BottomBarComponent.ctorParameters = () => [];
