// @smartstock/core-libs/esm2015/components/dash-card.component.js
import { Component, Input } from '@angular/core';
import { Router } from '@angular/router';
export class DashCardComponent {
    constructor(router) {
        this.router = router;
        this.title = '';
        this.height = 300;
        this.description = '';
    }
    ngOnInit() {
    }
    goTo() {
        if (this.reportLink) {
            this.router.navigateByUrl(this.reportLink).catch(console.log);
        }
    }
}
DashCardComponent.decorators = [
    { type: Component, args: [{
                selector: 'smartstock-dash-card',
                template: `
    <mat-card class="d-flex flex-column" [ngStyle]="{height:height+'px'}">
      <mat-card-header style="margin: 0">
        <span>{{title}}</span>
        <span style="margin-left: 8px"></span>
        <mat-icon matTooltip="{{description}}">info_outline</mat-icon>
      </mat-card-header>
      <mat-card-content class="flex-grow-1">
        <ng-container [ngTemplateOutlet]="content"></ng-container>
      </mat-card-content>
      <hr *ngIf="reportLink" style="width: 100%; margin: 0; padding: 0">
      <mat-card-actions *ngIf="reportLink" class="d-flex flex-row">
        <span style="flex: 1 1 auto"></span>
        <button (click)="goTo()"color="primary" mat-button>
          EXPLORE REPORT
          <mat-icon matSuffix>arrow_forward</mat-icon>
        </button>
      </mat-card-actions>
    </mat-card>
  `
            },] }
];
DashCardComponent.ctorParameters = () => [
    { type: Router }
];
DashCardComponent.propDecorators = {
    title: [{ type: Input }],
    height: [{ type: Input }],
    description: [{ type: Input }],
    reportLink: [{ type: Input }],
    content: [{ type: Input }]
};
