// @smartstock/core-libs/esm2015/components/data-not-ready.component.js
import { Component, Input } from '@angular/core';
export class DataNotReadyComponent {
    constructor() {
        this.width = 200;
        this.height = 200;
        this.isLoading = false;
    }
    ngOnInit() {
    }
}
DataNotReadyComponent.decorators = [
    { type: Component, args: [{
                selector: 'smartstock-data-not-ready',
                template: `
    <div *ngIf="!isLoading" style="margin-top: 10px; margin-bottom: 10px"
         class="d-flex flex-column justify-content-center align-items-center">
      <img alt="no data" src="../../../../assets/img/data.svg" width="{{width}}" height="{{height}}"/>
      <span>Data is not ready yet</span>
    </div>
    <div *ngIf="isLoading" style="width: 100%; height: 100%; display: flex; justify-content: center; align-items: center">
      <mat-progress-spinner mode="indeterminate" [diameter]="30"></mat-progress-spinner>
    </div>
  `
            },] }
];
DataNotReadyComponent.ctorParameters = () => [];
DataNotReadyComponent.propDecorators = {
    width: [{ type: Input }],
    height: [{ type: Input }],
    isLoading: [{ type: Input }]
};
