// @smartstock/core-libs/esm2015/components/search-input.component.js
import { Component, Input } from '@angular/core';
export class SearchInputComponent {
    constructor() {
        this.showSearch = false;
        this.searchProgressFlag = false;
        this.searchPlaceholder = 'Enter something...';
    }
    ngOnInit() {
    }
}
SearchInputComponent.decorators = [
    { type: Component, args: [{
                selector: 'smartstock-search-input',
                template: `
    <div style="width: 100%">
      <input [formControl]="searchInputControl"
             *ngIf="showSearch"
             class="toolbar-input" type="text" placeholder="{{searchPlaceholder}}">
      <mat-progress-spinner style="display: inline-block; margin-left: -30px"
                            mode="indeterminate" diameter="25"
                            color="primary"
                            *ngIf="showSearch && searchProgressFlag"
                            matSuffix>
        <!--      <mat-icon>search</mat-icon>-->
      </mat-progress-spinner>
    </div>
  `,
                styles: [".toolbar-input{background:hsla(0,0%,96.5%,.76);border:none;border-radius:4px;box-shadow:none;color:#000;outline:none;padding:8px;width:100%}.toolbar-input::-moz-placeholder{color:#535353}.toolbar-input:-ms-input-placeholder{color:#535353}.toolbar-input::placeholder{color:#535353}.toolbar-input:focus{outline:none}"]
            },] }
];
SearchInputComponent.ctorParameters = () => [];
SearchInputComponent.propDecorators = {
    searchInputControl: [{ type: Input }],
    showSearch: [{ type: Input }],
    searchProgressFlag: [{ type: Input }],
    searchPlaceholder: [{ type: Input }]
};
