// @smartstock/core-libs/bundles/smartstock-core-libs.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('rxjs'), require('@angular/material/dialog'), require('@angular/common/http'), require('bfastjs'), require('uuid'), require('@angular/router'), require('@angular/forms'), require('rxjs/operators'), require('@angular/material/button'), require('@angular/material/card'), require('@angular/material/divider'), require('@angular/material/expansion'), require('@angular/material/form-field'), require('@angular/material/icon'), require('@angular/material/list'), require('@angular/material/menu'), require('@angular/material/progress-bar'), require('@angular/material/progress-spinner'), require('@angular/material/select'), require('@angular/material/toolbar'), require('@angular/material/badge'), require('@angular/material/tooltip'), require('@angular/material/core'), require('@angular/common'), require('@angular/material/snack-bar')) :
    typeof define === 'function' && define.amd ? define('@smartstock/core-libs', ['exports', '@angular/core', 'rxjs', '@angular/material/dialog', '@angular/common/http', 'bfastjs', 'uuid', '@angular/router', '@angular/forms', 'rxjs/operators', '@angular/material/button', '@angular/material/card', '@angular/material/divider', '@angular/material/expansion', '@angular/material/form-field', '@angular/material/icon', '@angular/material/list', '@angular/material/menu', '@angular/material/progress-bar', '@angular/material/progress-spinner', '@angular/material/select', '@angular/material/toolbar', '@angular/material/badge', '@angular/material/tooltip', '@angular/material/core', '@angular/common', '@angular/material/snack-bar'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.smartstock = global.smartstock || {}, global.smartstock['core-libs'] = {}), global.ng.core, global.rxjs, global.ng.material.dialog, global.ng.common.http, global.bfastjs, global.uuid, global.ng.router, global.ng.forms, global.rxjs.operators, global.ng.material.button, global.ng.material.card, global.ng.material.divider, global.ng.material.expansion, global.ng.material.formField, global.ng.material.icon, global.ng.material.list, global.ng.material.menu, global.ng.material.progressBar, global.ng.material.progressSpinner, global.ng.material.select, global.ng.material.toolbar, global.ng.material.badge, global.ng.material.tooltip, global.ng.material.core, global.ng.common, global.ng.material.snackBar));
}(this, (function (exports, i0, rxjs, i3, i1, bfastjs, uuid, router, forms, operators, button, card, divider, expansion, formField, icon, list, menu, progressBar, progressSpinner, select, toolbar, badge, tooltip, core, common, i1$1) { 'use strict';

    var EventService = /** @class */ (function () {
        function EventService() {
        }
        EventService.prototype.broadcast = function (eventName, data) {
            var cE = new CustomEvent(eventName, {
                detail: data
            });
            window.dispatchEvent(cE);
        };
        EventService.prototype.listen = function (eventName, handler) {
            window.addEventListener(eventName, handler);
        };
        EventService.prototype.unListen = function (eventName, handler) {
            window.removeEventListener(eventName, handler);
        };
        return EventService;
    }());
    EventService.ɵprov = i0.ɵɵdefineInjectable({ factory: function EventService_Factory() { return new EventService(); }, token: EventService, providedIn: "root" });
    EventService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    EventService.ctorParameters = function () { return []; };

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    ;
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }
    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    var LogService = /** @class */ (function () {
        function LogService() {
        }
        LogService.prototype.i = function (message, tag, debug) {
            if (debug === void 0) { debug = false; }
            if (debug) {
                console.log(tag, message);
            }
        };
        LogService.prototype.e = function (message, tag, debug) {
            if (debug === void 0) { debug = false; }
            if (debug) {
                console.error(tag, message);
            }
        };
        LogService.prototype.w = function (message, tag, debug) {
            if (debug === void 0) { debug = false; }
            if (debug) {
                console.warn(tag, message);
            }
        };
        return LogService;
    }());
    LogService.ɵprov = i0.ɵɵdefineInjectable({ factory: function LogService_Factory() { return new LogService(); }, token: LogService, providedIn: "root" });
    LogService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    LogService.ctorParameters = function () { return []; };

    var SecurityUtil = /** @class */ (function () {
        function SecurityUtil() {
        }
        SecurityUtil.generateUUID = function () {
            return uuid.v4();
        };
        SecurityUtil.randomString = function (length) {
            var result = '';
            var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
            var charactersLength = characters.length;
            for (var i = 0; i < length; i++) {
                result += characters.charAt(Math.floor(Math.random() * charactersLength));
            }
            return result;
        };
        return SecurityUtil;
    }());

    var SsmEvents = {
        ACTIVE_SHOP_SET: '_ssm_active_shop_set_',
        ACTIVE_SHOP_REMOVE: '_ssm_active_shop_removed_',
        SETTINGS_UPDATED: '_ssm_settings_updated_',
        STOCK_UPDATED: '_ssm_stock_updated_',
        ADD_CART: 'add_cart',
        NO_OF_CART: 'noofProductsCart',
        CART_ITEMS: '_cart_items_'
    };

    var StorageService = /** @class */ (function () {
        function StorageService(eventApi) {
            this.eventApi = eventApi;
            this.smartStockCache = bfastjs.BFast.cache({ database: 'smartstock', collection: 'config' });
        }
        StorageService.prototype.getActiveUser = function () {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, bfastjs.BFast.auth().currentUser()];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        StorageService.prototype.saveSales = function (batchs) {
            return __awaiter(this, void 0, void 0, function () {
                var activeShop;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.getActiveShop()];
                        case 1:
                            activeShop = _a.sent();
                            return [4 /*yield*/, bfastjs.BFast.cache({ database: 'sales', collection: activeShop.projectId })
                                    .set(SecurityUtil.randomString(12), batchs, {
                                    dtl: 720
                                })];
                        case 2:
                            _a.sent();
                            return [2 /*return*/];
                    }
                });
            });
        };
        StorageService.prototype.getActiveShop = function () {
            return __awaiter(this, void 0, void 0, function () {
                var response;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.smartStockCache.get('activeShop')];
                        case 1:
                            response = _a.sent();
                            if (response) {
                                return [2 /*return*/, response];
                            }
                            else {
                                throw { message: 'No Active Shop' };
                            }
                            return [2 /*return*/];
                    }
                });
            });
        };
        StorageService.prototype.saveActiveShop = function (shop) {
            return __awaiter(this, void 0, void 0, function () {
                var response;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.smartStockCache.set('activeShop', shop, {
                                dtl: 7
                            })];
                        case 1:
                            response = _a.sent();
                            this.eventApi.broadcast(SsmEvents.ACTIVE_SHOP_SET);
                            return [2 /*return*/, response];
                    }
                });
            });
        };
        StorageService.prototype.getCurrentProjectId = function () {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.smartStockCache.get('cPID')];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        StorageService.prototype.saveCurrentProjectId = function (projectId) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.smartStockCache.set('cPID', projectId, {
                                dtl: 7
                            })];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        StorageService.prototype.clearSmartStockCache = function () {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.smartStockCache.clearAll()];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        StorageService.prototype.saveActiveUser = function (user) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    return [2 /*return*/, bfastjs.BFast.auth().setCurrentUser(user, 6)];
                });
            });
        };
        StorageService.prototype.removeActiveShop = function () {
            return __awaiter(this, void 0, void 0, function () {
                var response;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.smartStockCache.set('activeShop', undefined)];
                        case 1:
                            response = _a.sent();
                            this.eventApi.broadcast(SsmEvents.ACTIVE_SHOP_REMOVE);
                            return [2 /*return*/, response];
                    }
                });
            });
        };
        StorageService.prototype.removeActiveUser = function () {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, bfastjs.BFast.auth().setCurrentUser(undefined, 0)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        StorageService.prototype.removeStocks = function () {
            return __awaiter(this, void 0, void 0, function () {
                var shop;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.getActiveShop()];
                        case 1:
                            shop = _a.sent();
                            return [4 /*yield*/, bfastjs.BFast.cache({ database: 'stocks', collection: shop.projectId }).clearAll()];
                        case 2: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        StorageService.prototype.getStocks = function () {
            return __awaiter(this, void 0, void 0, function () {
                var shop, stocksCache;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.getActiveShop()];
                        case 1:
                            shop = _a.sent();
                            stocksCache = bfastjs.BFast.cache({ database: 'stocks', collection: shop.projectId });
                            return [4 /*yield*/, stocksCache.get('all')];
                        case 2: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        StorageService.prototype.saveStocks = function (stocks) {
            return __awaiter(this, void 0, void 0, function () {
                var shop, stocksCache;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.getActiveShop()];
                        case 1:
                            shop = _a.sent();
                            stocksCache = bfastjs.BFast.cache({ database: 'stocks', collection: shop.projectId });
                            return [4 /*yield*/, stocksCache.set('all', stocks, {
                                    dtl: 360
                                })];
                        case 2: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        StorageService.prototype.saveStock = function (stock) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    // const shop = await this.getActiveShop();
                    // const stocksCache = BFast.cache({database: 'stocks', collection: shop.projectId});
                    // return stocksCache.set(stock.id, stock);
                    return [2 /*return*/, undefined];
                });
            });
        };
        StorageService.prototype.getCustomers = function () {
            return __awaiter(this, void 0, void 0, function () {
                var shop, customersCache, customersKey, customers, customersKey_1, customersKey_1_1, key, _a, _b, e_1_1;
                var e_1, _c;
                return __generator(this, function (_d) {
                    switch (_d.label) {
                        case 0: return [4 /*yield*/, this.getActiveShop()];
                        case 1:
                            shop = _d.sent();
                            customersCache = bfastjs.BFast.cache({ database: 'customers', collection: shop.projectId });
                            return [4 /*yield*/, customersCache.keys()];
                        case 2:
                            customersKey = _d.sent();
                            customers = [];
                            _d.label = 3;
                        case 3:
                            _d.trys.push([3, 8, 9, 10]);
                            customersKey_1 = __values(customersKey), customersKey_1_1 = customersKey_1.next();
                            _d.label = 4;
                        case 4:
                            if (!!customersKey_1_1.done) return [3 /*break*/, 7];
                            key = customersKey_1_1.value;
                            _b = (_a = customers).push;
                            return [4 /*yield*/, customersCache.get(key)];
                        case 5:
                            _b.apply(_a, [_d.sent()]);
                            _d.label = 6;
                        case 6:
                            customersKey_1_1 = customersKey_1.next();
                            return [3 /*break*/, 4];
                        case 7: return [3 /*break*/, 10];
                        case 8:
                            e_1_1 = _d.sent();
                            e_1 = { error: e_1_1 };
                            return [3 /*break*/, 10];
                        case 9:
                            try {
                                if (customersKey_1_1 && !customersKey_1_1.done && (_c = customersKey_1.return)) _c.call(customersKey_1);
                            }
                            finally { if (e_1) throw e_1.error; }
                            return [7 /*endfinally*/];
                        case 10: return [2 /*return*/, customers];
                    }
                });
            });
        };
        StorageService.prototype.saveCustomer = function (customer) {
            return __awaiter(this, void 0, void 0, function () {
                var shop, customersCache;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.getActiveShop()];
                        case 1:
                            shop = _a.sent();
                            customersCache = bfastjs.BFast.cache({ database: 'customers', collection: shop.projectId });
                            return [4 /*yield*/, customersCache.set(customer.displayName, customer, {
                                    dtl: 360
                                })];
                        case 2: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        return StorageService;
    }());
    StorageService.ɵprov = i0.ɵɵdefineInjectable({ factory: function StorageService_Factory() { return new StorageService(i0.ɵɵinject(EventService)); }, token: StorageService, providedIn: "root" });
    StorageService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    StorageService.ctorParameters = function () { return [
        { type: EventService }
    ]; };

    var SettingsService = /** @class */ (function () {
        function SettingsService(httpClient, storageService, indexDb) {
            this.httpClient = httpClient;
            this.storageService = storageService;
            this.indexDb = indexDb;
            this.ssmServerURL = 'https://smartstock-daas.bfast.fahamutech.com'; // environment.smartstock.databaseURL;
            this.ssmFunctionsURL = 'https://smartstock-faas.bfast.fahamutech.com'; // environment.smartstock.functionsURL;
            this.ssmHeader = {
                'X-Parse-Application-Id': 'smartstock'
            };
            this.ssmFunctionsHeader = {
                'bfast-application-id': 'smartstock',
                'content-type': 'application/json'
            };
        }
        SettingsService.prototype.getSSMUserHeader = function () {
            return __awaiter(this, void 0, void 0, function () {
                var user, activeShop, e_1;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 3, , 4]);
                            return [4 /*yield*/, this.storageService.getActiveUser()];
                        case 1:
                            user = _a.sent();
                            return [4 /*yield*/, this.storageService.getActiveShop()];
                        case 2:
                            activeShop = _a.sent();
                            if (!user) {
                                // console.log('no user records found');
                                throw new Error('no user records found');
                            }
                            if (user && user.sessionToken && activeShop && activeShop.applicationId) {
                                return [2 /*return*/, {
                                        'X-Parse-Application-Id': 'smartstock',
                                        'X-Parse-Session-Token': user.sessionToken,
                                        'Content-Type': 'application/json'
                                    }];
                            }
                            else {
                                throw new Error('token not found');
                            }
                            return [3 /*break*/, 4];
                        case 3:
                            e_1 = _a.sent();
                            throw { message: 'Fails to get user, so to retrieve token' };
                        case 4: return [2 /*return*/];
                    }
                });
            });
        };
        SettingsService.prototype.getCustomerApplicationId = function () {
            return __awaiter(this, void 0, void 0, function () {
                var activeShop, e_2;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.storageService.getActiveShop()];
                        case 1:
                            activeShop = _a.sent();
                            if (!activeShop) {
                                throw new Error('No user record');
                            }
                            return [2 /*return*/, activeShop.applicationId];
                        case 2:
                            e_2 = _a.sent();
                            throw { message: 'Fails to get application id', reason: e_2.toString() };
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        SettingsService.prototype.getCustomerServerURLId = function () {
            return __awaiter(this, void 0, void 0, function () {
                var activeShop, reason_1;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.storageService.getActiveShop()];
                        case 1:
                            activeShop = _a.sent();
                            if (!activeShop) {
                                throw new Error('No user in local storage');
                            }
                            return [2 /*return*/, activeShop.projectUrlId];
                        case 2:
                            reason_1 = _a.sent();
                            throw { message: 'Fails to get user', reason: reason_1.toString() };
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        SettingsService.prototype.getCustomerHeader = function () {
            return __awaiter(this, void 0, void 0, function () {
                var _a, _b, e_3;
                return __generator(this, function (_c) {
                    switch (_c.label) {
                        case 0:
                            _c.trys.push([0, 2, , 3]);
                            _a = {};
                            _b = 'X-Parse-Application-Id';
                            return [4 /*yield*/, this.getCustomerApplicationId()];
                        case 1: return [2 /*return*/, (_a[_b] = _c.sent(),
                                _a)];
                        case 2:
                            e_3 = _c.sent();
                            console.warn(e_3);
                            return [2 /*return*/, {}];
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        SettingsService.prototype.getCustomerPostHeader = function (contentType) {
            return __awaiter(this, void 0, void 0, function () {
                var _a, _b, e_4;
                return __generator(this, function (_c) {
                    switch (_c.label) {
                        case 0:
                            _c.trys.push([0, 2, , 3]);
                            _a = {};
                            _b = 'X-Parse-Application-Id';
                            return [4 /*yield*/, this.getCustomerApplicationId()];
                        case 1: return [2 /*return*/, (_a[_b] = _c.sent(),
                                _a['content-type'] = contentType ? contentType : 'application/json',
                                _a)];
                        case 2:
                            e_4 = _c.sent();
                            throw { message: 'Fails to get customer post header', reason: e_4.toString() };
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        SettingsService.prototype.getCustomerServerURL = function () {
            return __awaiter(this, void 0, void 0, function () {
                var _a, e_5;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            _b.trys.push([0, 2, , 3]);
                            _a = "https://";
                            return [4 /*yield*/, this.getCustomerServerURLId()];
                        case 1: return [2 /*return*/, _a + (_b.sent()) + ".bfast.fahamutech.com"];
                        case 2:
                            e_5 = _b.sent();
                            throw { message: 'Fails to get server url', reason: e_5.toString() };
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        SettingsService.prototype.getCustomerProjectId = function () {
            return __awaiter(this, void 0, void 0, function () {
                var activeShop, e_6;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.indexDb.getActiveShop()];
                        case 1:
                            activeShop = _a.sent();
                            if (!activeShop) {
                                throw new Error('No user in local storage');
                            }
                            return [2 /*return*/, activeShop.projectId];
                        case 2:
                            e_6 = _a.sent();
                            throw { message: 'Fails to get project id', reason: e_6.toString() };
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        /**
         * @deprecated
         */
        SettingsService.prototype.getPrinterAddress = function (callback) {
            // this.indexDb.getItem<{ ip: string, name: string }>('printerAddress').then(value => {
            //   callback(null);
            // }).catch(reason => {
            //   console.log(reason);
            //   callback(null);
            // });
            callback(null);
        };
        SettingsService.prototype.saveSettings = function (settings) {
            var _this = this;
            return new Promise(function (resolve, reject) { return __awaiter(_this, void 0, void 0, function () {
                var activeShop_1, e_7;
                var _this = this;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.storageService.getActiveShop()];
                        case 1:
                            activeShop_1 = _a.sent();
                            this.httpClient.put(this.ssmFunctionsURL + '/settings/' + activeShop_1.projectId, settings, {
                                headers: this.ssmFunctionsHeader
                            }).subscribe(function (_) {
                                activeShop_1.settings = _.settings;
                                _this.storageService.saveActiveShop(activeShop_1).then(function (_1) {
                                    resolve('Shop settings updated');
                                }).catch(function (reason) {
                                    reject(reason);
                                });
                            }, function (error) {
                                reject(error);
                            });
                            return [3 /*break*/, 3];
                        case 2:
                            e_7 = _a.sent();
                            reject(e_7);
                            return [3 /*break*/, 3];
                        case 3: return [2 /*return*/];
                    }
                });
            }); });
        };
        SettingsService.prototype.getSettings = function () {
            return __awaiter(this, void 0, void 0, function () {
                var activeShop, e_8;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.storageService.getActiveShop()];
                        case 1:
                            activeShop = _a.sent();
                            if (!activeShop || !activeShop.settings) {
                                return [2 /*return*/, {
                                        printerFooter: 'Thank you',
                                        printerHeader: '',
                                        saleWithoutPrinter: true,
                                        allowRetail: true,
                                        allowWholesale: true,
                                    }];
                            }
                            return [2 /*return*/, activeShop.settings];
                        case 2:
                            e_8 = _a.sent();
                            throw { message: 'Fails to get settings', reason: e_8.toString() };
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        return SettingsService;
    }());
    SettingsService.ɵprov = i0.ɵɵdefineInjectable({ factory: function SettingsService_Factory() { return new SettingsService(i0.ɵɵinject(i1.HttpClient), i0.ɵɵinject(StorageService), i0.ɵɵinject(StorageService)); }, token: SettingsService, providedIn: "root" });
    SettingsService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    SettingsService.ctorParameters = function () { return [
        { type: i1.HttpClient },
        { type: StorageService },
        { type: StorageService }
    ]; };

    var VerifyEMailDialogComponent = /** @class */ (function () {
        function VerifyEMailDialogComponent(dialogRef) {
            this.dialogRef = dialogRef;
        }
        return VerifyEMailDialogComponent;
    }());
    VerifyEMailDialogComponent.decorators = [
        { type: i0.Component, args: [{
                    template: "\n    <div>\n      <div mat-dialog-title>Account Verification</div>\n      <div mat-dialog-content>\n        <p>\n          Your email is not verified we send a link to the email\n          <br>\n          you use when open account for you to verify your account\n        </p>\n      </div>\n      <div mat-dialog-actions>\n        <button (click)=\"dialogRef.close()\" mat-button color=\"primary\">Close</button>\n      </div>\n    </div>\n  "
                },] }
    ];
    VerifyEMailDialogComponent.ctorParameters = function () { return [
        { type: i3.MatDialogRef }
    ]; };

    var UserService = /** @class */ (function () {
        function UserService(httpClient, settingsService, dialog, logger, storageService) {
            this.httpClient = httpClient;
            this.settingsService = settingsService;
            this.dialog = dialog;
            this.logger = logger;
            this.storageService = storageService;
        }
        UserService.prototype.currentUser = function () {
            return __awaiter(this, void 0, void 0, function () {
                var user;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, bfastjs.BFast.auth().currentUser()];
                        case 1:
                            user = _a.sent();
                            if (!(user && user.role !== 'admin')) return [3 /*break*/, 2];
                            return [2 /*return*/, user];
                        case 2:
                            if (!(user && user.verified === true)) return [3 /*break*/, 3];
                            return [2 /*return*/, user];
                        case 3: return [4 /*yield*/, bfastjs.BFast.auth().setCurrentUser(undefined)];
                        case 4: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        UserService.prototype.deleteUser = function (user) {
            return __awaiter(this, void 0, void 0, function () {
                var _a, _b, _c, _d, _e;
                return __generator(this, function (_f) {
                    switch (_f.label) {
                        case 0:
                            _b = (_a = bfastjs.BFast.functions()
                                .request('/functions/users/' + user.id))
                                .delete;
                            _c = {};
                            _d = {};
                            _e = {};
                            return [4 /*yield*/, bfastjs.BFast.auth().currentUser()];
                        case 1: return [2 /*return*/, _b.apply(_a, [(_c.data = (_d.context = (_e.admin = _f.sent(), _e), _d),
                                    _c)])];
                    }
                });
            });
        };
        UserService.prototype.getAllUser = function (pagination) {
            return __awaiter(this, void 0, void 0, function () {
                var projectId;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.settingsService.getCustomerProjectId()];
                        case 1:
                            projectId = _a.sent();
                            return [2 /*return*/, bfastjs.BFast.database().collection('_User')
                                    .query()
                                    .equalTo('projectId', projectId)
                                    .includesIn('role', ['user', 'manager'])
                                    .size(pagination.size)
                                    .skip(pagination.skip)
                                    .find({
                                    useMasterKey: true
                                })];
                    }
                });
            });
        };
        UserService.prototype.getUser = function (user, callback) {
        };
        UserService.prototype.login = function (user) {
            return __awaiter(this, void 0, void 0, function () {
                var authUser;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, bfastjs.BFast.auth().logIn(user.username, user.password)];
                        case 1:
                            authUser = _a.sent();
                            return [4 /*yield*/, this.storageService.removeActiveShop()];
                        case 2:
                            _a.sent();
                            if (!(authUser && authUser.role !== 'admin')) return [3 /*break*/, 4];
                            return [4 /*yield*/, this.storageService.saveActiveUser(authUser)];
                        case 3:
                            _a.sent();
                            return [2 /*return*/, authUser];
                        case 4:
                            if (!(authUser && authUser.verified === true)) return [3 /*break*/, 6];
                            return [4 /*yield*/, this.storageService.saveActiveUser(authUser)];
                        case 5:
                            _a.sent();
                            return [2 /*return*/, authUser];
                        case 6: return [4 /*yield*/, bfastjs.BFast.functions().request('/functions/users/reVerifyAccount/' + user.username).post()];
                        case 7:
                            _a.sent();
                            this.dialog.open(VerifyEMailDialogComponent, {
                                closeOnNavigation: true,
                                disableClose: true
                            });
                            throw { code: 403, err: 'account not verified' };
                    }
                });
            });
        };
        UserService.prototype.logout = function (user) {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, bfastjs.BFast.auth().logOut()];
                        case 1:
                            _a.sent();
                            return [4 /*yield*/, this.storageService.removeActiveUser()];
                        case 2:
                            _a.sent();
                            return [4 /*yield*/, this.storageService.removeActiveShop()];
                        case 3:
                            _a.sent();
                            return [2 /*return*/];
                    }
                });
            });
        };
        UserService.prototype.register = function (user) {
            return __awaiter(this, void 0, void 0, function () {
                var e_1;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 3, , 4]);
                            user.settings = {
                                printerFooter: 'Thank you',
                                printerHeader: '',
                                saleWithoutPrinter: true,
                                allowRetail: true,
                                allowWholesale: true
                            };
                            user.shops = [];
                            return [4 /*yield*/, this.storageService.removeActiveShop()];
                        case 1:
                            _a.sent();
                            return [4 /*yield*/, bfastjs.BFast.functions().request('/functions/users/create').post(user, {
                                    headers: this.settingsService.ssmFunctionsHeader
                                })];
                        case 2: return [2 /*return*/, _a.sent()];
                        case 3:
                            e_1 = _a.sent();
                            if (e_1 && e_1.response && e_1.response.data) {
                                throw e_1.response.data;
                            }
                            else {
                                throw e_1.toString();
                            }
                            return [3 /*break*/, 4];
                        case 4: return [2 /*return*/];
                    }
                });
            });
        };
        UserService.prototype.resetPassword = function (username) {
            return bfastjs.BFast.functions().request('/functions/users/resetPassword/' + encodeURIComponent(username)).get();
        };
        UserService.prototype.refreshToken = function () {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    return [2 /*return*/, bfastjs.BFast.auth().currentUser()];
                });
            });
        };
        UserService.prototype.addUser = function (user) {
            var _this = this;
            return new Promise(function (resolve, reject) { return __awaiter(_this, void 0, void 0, function () {
                var shop, shops, shops1;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.storageService.getActiveShop()];
                        case 1:
                            shop = _a.sent();
                            shops = user.shops ? user.shops : [];
                            shops1 = shops.filter(function (value) { return value.applicationId !== shop.applicationId; });
                            user.applicationId = shop.applicationId;
                            user.projectUrlId = shop.projectUrlId;
                            user.projectId = shop.projectId;
                            user.businessName = shop.businessName;
                            user.settings = shop.settings;
                            user.shops = shops1;
                            this.httpClient.post(this.settingsService.ssmFunctionsURL + '/functions/users/seller', user, {
                                headers: this.settingsService.ssmFunctionsHeader
                            }).subscribe(function (value) {
                                resolve(value);
                            }, function (error) {
                                reject(error);
                            });
                            return [2 /*return*/];
                    }
                });
            }); });
        };
        UserService.prototype.getShops = function () {
            return __awaiter(this, void 0, void 0, function () {
                var user, shops_1, e_2;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.storageService.getActiveUser()];
                        case 1:
                            user = _a.sent();
                            shops_1 = [];
                            user.shops.forEach(function (element) {
                                shops_1.push(element);
                            });
                            shops_1.push({
                                businessName: user.businessName,
                                projectId: user.projectId,
                                applicationId: user.applicationId,
                                projectUrlId: user.projectUrlId,
                                settings: user.settings,
                                street: user.street,
                                country: user.country,
                                region: user.region
                            });
                            return [2 /*return*/, shops_1];
                        case 2:
                            e_2 = _a.sent();
                            throw e_2;
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        UserService.prototype.getCurrentShop = function () {
            return __awaiter(this, void 0, void 0, function () {
                var activeShop, e_3;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.storageService.getActiveShop()];
                        case 1:
                            activeShop = _a.sent();
                            if (activeShop && activeShop.projectId && activeShop.applicationId && activeShop.projectUrlId) {
                                return [2 /*return*/, activeShop];
                            }
                            else {
                                throw new Error('No active shop in records');
                            }
                            return [3 /*break*/, 3];
                        case 2:
                            e_3 = _a.sent();
                            throw e_3;
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        UserService.prototype.saveCurrentShop = function (shop) {
            return __awaiter(this, void 0, void 0, function () {
                var e_4;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 3, , 4]);
                            return [4 /*yield*/, this.storageService.saveCurrentProjectId(shop.projectId)];
                        case 1:
                            _a.sent();
                            return [4 /*yield*/, this.storageService.saveActiveShop(shop)];
                        case 2: return [2 /*return*/, _a.sent()];
                        case 3:
                            e_4 = _a.sent();
                            throw e_4;
                        case 4: return [2 /*return*/];
                    }
                });
            });
        };
        UserService.prototype.createShop = function (data) {
            return undefined;
            // return new Promise<ShopModel>(async (resolve, reject) => {
            //   this.httpClient.post<ShopModel>(this.settings.ssmFunctionsURL + '/functions/shop', data, {
            //     headers: this.settings.ssmFunctionsHeader
            //   }).subscribe(value => {
            //     resolve(value);
            //   }, error => {
            //     reject(error);
            //   });
            // });
        };
        UserService.prototype.deleteShop = function (shop) {
            return undefined;
            // return new Promise<ShopModel>((resolve, reject) => {
            //   this.httpClient.delete(this.settings.ssmFunctionsURL + '/functions/shop', {
            //   })
            // });
        };
        UserService.prototype.updatePassword = function (user, password) {
            var _this = this;
            return new Promise(function (resolve, reject) {
                _this.httpClient.put(_this.settingsService.ssmFunctionsURL + '/functions/users/password/' + user.id, {
                    password: password
                }, {
                    headers: _this.settingsService.ssmFunctionsHeader
                }).subscribe(function (value) {
                    resolve(value);
                }, function (error) {
                    reject(error);
                });
            });
        };
        UserService.prototype.updateUser = function (user, data) {
            var _this = this;
            return new Promise(function (resolve, reject) { return __awaiter(_this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    this.httpClient.put(this.settingsService.ssmFunctionsURL + '/functions/users/' + user.id, data, {
                        headers: this.settingsService.ssmFunctionsHeader
                    }).subscribe(function (value) { return resolve(value); }, function (error1) {
                        reject(error1);
                    });
                    return [2 /*return*/];
                });
            }); });
        };
        UserService.prototype.updateCurrentUser = function (user) {
            return __awaiter(this, void 0, void 0, function () {
                var e_5;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.storageService.saveActiveUser(user)];
                        case 1: return [2 /*return*/, _a.sent()];
                        case 2:
                            e_5 = _a.sent();
                            throw e_5;
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        UserService.prototype.changePasswordFromOld = function (data) {
            var _this = this;
            return new Promise(function (resolve, reject) {
                _this.httpClient.put(_this.settingsService.ssmFunctionsURL + '/functions/users/password/change/' + data.user.id, {
                    lastPassword: data.lastPassword,
                    username: data.user.username,
                    password: data.password
                }, {
                    headers: _this.settingsService.ssmFunctionsHeader
                }).subscribe(function (value) { return resolve(value); }, function (error1) {
                    reject(error1);
                });
            });
        };
        return UserService;
    }());
    UserService.ɵprov = i0.ɵɵdefineInjectable({ factory: function UserService_Factory() { return new UserService(i0.ɵɵinject(i1.HttpClient), i0.ɵɵinject(SettingsService), i0.ɵɵinject(i3.MatDialog), i0.ɵɵinject(LogService), i0.ɵɵinject(StorageService)); }, token: UserService, providedIn: "root" });
    UserService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    UserService.ctorParameters = function () { return [
        { type: i1.HttpClient },
        { type: SettingsService },
        { type: i3.MatDialog },
        { type: LogService },
        { type: StorageService }
    ]; };

    var ConfigsService = /** @class */ (function () {
        function ConfigsService() {
        }
        return ConfigsService;
    }());
    ConfigsService.versionName = '';
    ConfigsService.ɵprov = i0.ɵɵdefineInjectable({ factory: function ConfigsService_Factory() { return new ConfigsService(); }, token: ConfigsService, providedIn: "root" });
    ConfigsService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];

    var DrawerComponent = /** @class */ (function () {
        function DrawerComponent(userService, logger, eventApi) {
            this.userService = userService;
            this.logger = logger;
            this.eventApi = eventApi;
            this.versionNumber = rxjs.of();
            this.selectedMenu = {
                background: '#1b5e20',
                borderTopRightRadius: '50px',
                borderBottomRightRadius: '50px',
                color: 'white'
            };
        }
        DrawerComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.versionNumber = rxjs.of(ConfigsService.versionName);
            this.userService.getCurrentShop().then(function (shop) {
                _this.shop = shop;
            }).catch(function (reason) {
                _this.logger.i(reason);
                _this.shop = undefined;
            });
            this.userService.currentUser().then(function (user) {
                _this.currentUser = user;
            });
            this.eventApi.listen(SsmEvents.SETTINGS_UPDATED, function (data) {
                _this.userService.getCurrentShop().then(function (shop) {
                    _this.shop = shop;
                }).catch(function (reason) {
                    _this.logger.e(reason, 'AdminDrawerComponent:37');
                    _this.shop = undefined;
                });
            });
        };
        DrawerComponent.prototype.shouldExpand = function (route) {
            var url = new URL(location.href);
            return url.pathname.startsWith('/' + route);
        };
        DrawerComponent.prototype.isManager = function () {
            return this.currentUser && (this.currentUser.role === 'admin' || this.currentUser.role === 'manager');
        };
        DrawerComponent.prototype.isAdmin = function () {
            return this.currentUser && (this.currentUser.role === 'admin');
        };
        return DrawerComponent;
    }());
    DrawerComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-drawer',
                    template: "\n    <div class=\"my-side-nav\">\n      <div>\n        <div class=\"d-flex justify-content-center align-items-center flex-column\" style=\"padding-bottom: 8px\">\n          <div style=\"padding: 16px; justify-content: center; align-items: center\">\n            <mat-icon style=\"width: 70px; height: 70px; font-size: 70px\" color=\"primary\">store</mat-icon>\n          </div>\n          <span style=\"max-width: 150px; overflow: hidden; text-overflow: ellipsis;\"\n                *ngIf=\"shop\">{{shop.businessName}}</span>\n          <span style=\"width: 4px; height: 4px\"></span>\n          <button style=\"width: 80%\"\n                  color=\"primary\"\n                  class=\"btn-block\"\n                  routerLink=\"/account/shop\" mat-button>\n            Change Shop\n          </button>\n        </div>\n\n        <div class=\"d-flex justify-content-center align-items-center\">\n          <mat-divider></mat-divider>\n        </div>\n\n        <mat-accordion [multi]=\"true\" class=\"mat-elevation-z0\">\n          <mat-nav-list *ngIf=\"isAdmin()\" class=\"mat-elevation-z0\">\n            <mat-list-item [ngStyle]=\"shouldExpand('dashboard')?selectedMenu:{}\" routerLink=\"/dashboard\">\n              <mat-icon matListIcon matPrefix>dashboard</mat-icon>\n              <span matLine style=\"margin-left: 8px\">Dashboard</span>\n            </mat-list-item>\n          </mat-nav-list>\n\n          <mat-divider *ngIf=\"isAdmin()\"></mat-divider>\n\n          <mat-expansion-panel *ngIf=\"isAdmin()\" [expanded]=\"shouldExpand('report')\" class=\"mat-elevation-z0\">\n            <mat-expansion-panel-header>\n              <mat-icon matPrefix>table_chart</mat-icon>\n              <span style=\"margin-left: 8px\">Reports</span>\n            </mat-expansion-panel-header>\n            <mat-nav-list>\n              <a mat-list-item routerLink=\"/report/sales\">Sales Reports</a>\n              <a mat-list-item routerLink=\"/report/stocks\">Stock Reports</a>\n            </mat-nav-list>\n          </mat-expansion-panel>\n\n          <mat-divider *ngIf=\"currentUser && currentUser.role==='admin'\"></mat-divider>\n\n          <mat-nav-list>\n            <mat-list-item [ngStyle]=\"shouldExpand('sale')?selectedMenu:{}\" routerLink=\"/sale\">\n              <mat-icon matListIcon matPrefix>shop_front</mat-icon>\n              <span matLine style=\"margin-left: 8px\">Sale</span>\n            </mat-list-item>\n          </mat-nav-list>\n\n          <mat-divider></mat-divider>\n\n          <mat-expansion-panel *ngIf=\"isManager()\" [expanded]=\"shouldExpand('purchase')\" class=\"mat-elevation-z0\">\n            <mat-expansion-panel-header>\n              <mat-icon matPrefix>receipts</mat-icon>\n              <span style=\"margin-left: 8px\">Purchase</span>\n            </mat-expansion-panel-header>\n            <mat-nav-list>\n              <a mat-list-item routerLink=\"/purchase\">All Purchases</a>\n              <a mat-list-item routerLink=\"/purchase/create\">New Purchase</a>\n              <a mat-list-item routerLink=\"/stock\" [queryParams]=\"{t:3}\">Suppliers</a>\n            </mat-nav-list>\n          </mat-expansion-panel>\n\n          <mat-divider *ngIf=\"currentUser && (currentUser.role==='admin' || currentUser.role === 'manager')\"></mat-divider>\n\n          <mat-nav-list *ngIf=\"isManager()\">\n            <mat-list-item [ngStyle]=\"shouldExpand('stock')?selectedMenu:{}\" routerLink=\"/stock\">\n              <mat-icon matListIcon matPrefix>store</mat-icon>\n              <span matLine style=\"margin-left: 8px\">Stock</span>\n            </mat-list-item>\n          </mat-nav-list>\n\n          <mat-divider *ngIf=\"isManager()\"></mat-divider>\n\n          <mat-expansion-panel [expanded]=\"shouldExpand('settings')\" class=\"mat-elevation-z0\">\n            <mat-expansion-panel-header>\n              <mat-icon matPrefix>supervisor_account</mat-icon>\n              <span style=\"margin-left: 8px\">Account</span>\n            </mat-expansion-panel-header>\n            <mat-nav-list>\n              <a *ngIf=\"isManager()\" mat-list-item\n                 routerLink=\"/account/settings\">\n                <div class=\"d-flex flex-row flex-nowrap btn-block\">\n                  <span>Settings</span>\n                  <span class=\"flex-grow-1\"></span>\n                  <mat-icon>settings</mat-icon>\n                </div>\n              </a>\n              <a *ngIf=\"isManager()\" mat-list-item\n                 routerLink=\"/account/users\">\n                <div class=\"d-flex flex-row flex-nowrap btn-block\">\n                  <span>Users</span>\n                  <span class=\"flex-grow-1\"></span>\n                  <mat-icon>person_add</mat-icon>\n                </div>\n              </a>\n              <a mat-list-item routerLink=\"/account/profile\">\n                <div class=\"d-flex flex-row flex-nowrap btn-block\">\n                  <span>Profile</span>\n                  <span class=\"flex-grow-1\"></span>\n                  <mat-icon>person</mat-icon>\n                </div>\n              </a>\n            </mat-nav-list>\n          </mat-expansion-panel>\n        </mat-accordion>\n\n      </div>\n      <span style=\"flex-grow: 1\"></span>\n      <div style=\"display: flex; justify-content: center; align-items: center\">\n        <span>Version: {{versionNumber | async}}</span>\n      </div>\n    </div>\n  ",
                    styles: [".my-side-nav{display:flex;flex-direction:column;min-height:100vh;width:100%;z-index:50000}.drawer-head{background-position:50%;background-repeat:no-repeat;background-size:cover;height:180px}div.mat-list-item-content{width:100%!important}mat-divider{margin-left:10%;margin-right:10%;width:80%}"]
                },] }
    ];
    DrawerComponent.ctorParameters = function () { return [
        { type: UserService },
        { type: LogService },
        { type: EventService }
    ]; };
    DrawerComponent.propDecorators = {
        versionNumber: [{ type: i0.Input }]
    };

    var ToolbarComponent = /** @class */ (function () {
        function ToolbarComponent(router, storage, userDatabase, eventService) {
            this.router = router;
            this.storage = storage;
            this.userDatabase = userDatabase;
            this.eventService = eventService;
            this.showProgress = false;
            this.hasBackRoute = false;
            this.showSearch = false;
            this.searchCallback = new i0.EventEmitter();
            this.searchInputControl = new forms.FormControl('');
            this.searchProgressFlag = false;
            this.isMobile = false; // environment.android;
        }
        ToolbarComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.storage.getActiveUser().then(function (user) {
                _this.currentUser = user;
            });
            this.searchInputControl.valueChanges.pipe(operators.debounceTime(500), operators.distinctUntilChanged()).subscribe(function (_) {
                _this.searchCallback.emit(_this.searchInputControl.value);
            });
            this.getProductsInCart();
            this._clearSearchInputListener();
        };
        ToolbarComponent.prototype.getProductsInCart = function () {
            var _this = this;
            this.eventService.listen(SsmEvents.NO_OF_CART, function (data) {
                _this.noOfProductsInCart = data.detail;
            });
        };
        ToolbarComponent.prototype.logout = function () {
            var _this = this;
            this.userDatabase.logout(null).finally(function () {
                return _this.router.navigateByUrl('');
            }).catch(function (err) { return console.log(''); });
        };
        ToolbarComponent.prototype._clearSearchInputListener = function () {
            // this.eventService.listen(SsmEvents.ADD_CART, data => {
            //   this.searchInputControl.reset('');
            // });
        };
        return ToolbarComponent;
    }());
    ToolbarComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-toolbar',
                    template: "\n    <mat-toolbar color=\"primary\" class=\"sticky-top mat-elevation-z4\">\n      <mat-toolbar-row>\n<!--        <button routerLink=\"{{backLink}}\" *ngIf=\"hasBackRoute && backLink\" mat-icon-button>-->\n<!--          <mat-icon>arrow_back</mat-icon>-->\n<!--        </button>-->\n        <button mat-icon-button (click)=\"sidenav.toggle()\">\n          <mat-icon>menu</mat-icon>\n        </button>\n        <span>{{heading}}</span>\n        <span *ngIf=\"isMobile\" style=\"flex: 1 1 auto\"></span>\n        <span *ngIf=\"!isMobile && showSearch\" style=\"width: 16px\"></span>\n        <span *ngIf=\"!isMobile && !showSearch\" style=\"flex: 1 1 auto\"></span>\n        <smartstock-search-input [searchProgressFlag]=\"searchProgressFlag\"\n                          *ngIf=\"!isMobile && showSearch\" style=\"flex: 1 1 auto\"\n                          [showSearch]=\"showSearch\"\n                          [searchInputControl]=\"searchInputControl\"\n                          [searchPlaceholder]=\"searchPlaceholder\">\n        </smartstock-search-input>\n        <span *ngIf=\"!isMobile && showSearch\" style=\"width: 16px\"></span>\n        <button *ngIf=\"noOfProductsInCart> 0 && !isMobile\" mat-icon-button (click)=\"cartdrawer.toggle()\"\n                [matBadge]=\"noOfProductsInCart\">\n          <mat-icon>shopping_cart</mat-icon>\n        </button>\n        <button *ngIf=\"!isMobile\" class=\"ft-button\" mat-button [matMenuTriggerFor]=\"menu\">\n          <mat-icon>account_circle</mat-icon>\n          <span *ngIf=\"currentUser\">{{\"  \" + currentUser.username}}</span>\n        </button>\n        <button *ngIf=\"isMobile\" mat-icon-button [matMenuTriggerFor]=\"menu\">\n          <mat-icon>more_vert</mat-icon>\n          <!--      <span *ngIf=\"currentUser\">{{\"  \" + currentUser.username}}</span>-->\n        </button>\n        <mat-menu #menu>\n          <button mat-menu-item (click)=\"logout()\">\n            <mat-icon>exit_to_app</mat-icon>\n            Logout\n          </button>\n          <button mat-menu-item routerLink=\"/account/profile\">\n            <mat-icon>person</mat-icon>\n            My Profile\n          </button>\n        </mat-menu>\n      </mat-toolbar-row>\n\n      <mat-toolbar-row *ngIf=\"isMobile && showSearch\">\n        <!--    <span style=\"flex-grow: 1\"></span>-->\n        <smartstock-search-input [searchProgressFlag]=\"searchProgressFlag\"\n                          style=\"flex: 1 1 auto\"\n                          [showSearch]=\"showSearch\"\n                          [searchInputControl]=\"searchInputControl\"\n                          [searchPlaceholder]=\"searchPlaceholder\">\n        </smartstock-search-input>\n        <!--    <span style=\"flex-grow: 1\"></span>-->\n      </mat-toolbar-row>\n    </mat-toolbar>\n  "
                },] }
    ];
    ToolbarComponent.ctorParameters = function () { return [
        { type: router.Router },
        { type: StorageService },
        { type: UserService },
        { type: EventService }
    ]; };
    ToolbarComponent.propDecorators = {
        heading: [{ type: i0.Input }],
        showProgress: [{ type: i0.Input }],
        sidenav: [{ type: i0.Input }],
        hasBackRoute: [{ type: i0.Input }],
        backLink: [{ type: i0.Input }],
        cartdrawer: [{ type: i0.Input }],
        showSearch: [{ type: i0.Input }],
        searchCallback: [{ type: i0.Output }],
        searchInputControl: [{ type: i0.Input }],
        searchPlaceholder: [{ type: i0.Input }],
        searchProgressFlag: [{ type: i0.Input }]
    };

    var NoStockDialogComponent = /** @class */ (function () {
        function NoStockDialogComponent(dialogRef, data) {
            this.dialogRef = dialogRef;
            this.data = data;
        }
        NoStockDialogComponent.prototype.ngOnInit = function () {
        };
        return NoStockDialogComponent;
    }());
    NoStockDialogComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-no-stock-dialog',
                    template: "\n    <div>\n      <h5>Hello!</h5>\n      <mat-card-subtitle mat-dialog-content>\n        StockModel is not available yet, add product or refresh your stock\n      </mat-card-subtitle>\n      <div mat-dialog-actions>\n        <button (click)=\"dialogRef.close()\" routerLink=\"/stock\" mat-flat-button color=\"primary\">Go To StockModel</button>\n        <button (click)=\"dialogRef.close()\" mat-flat-button color=\"warn\">Close</button>\n      </div>\n    </div>\n  "
                },] }
    ];
    NoStockDialogComponent.ctorParameters = function () { return [
        { type: i3.MatDialogRef },
        { type: undefined, decorators: [{ type: i0.Inject, args: [i3.MAT_DIALOG_DATA,] }] }
    ]; };

    var ShopsPipe = /** @class */ (function () {
        function ShopsPipe(userApi) {
            this.userApi = userApi;
        }
        ShopsPipe.prototype.transform = function (shops) {
            var args = [];
            for (var _i = 1; _i < arguments.length; _i++) {
                args[_i - 1] = arguments[_i];
            }
            return __awaiter(this, void 0, void 0, function () {
                var shop, e_1;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            _a.trys.push([0, 2, , 3]);
                            return [4 /*yield*/, this.userApi.getCurrentShop()];
                        case 1:
                            shop = _a.sent();
                            shops.push(shop);
                            return [2 /*return*/, shops.map(function (value) { return value.businessName; })];
                        case 2:
                            e_1 = _a.sent();
                            return [2 /*return*/, []];
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        return ShopsPipe;
    }());
    ShopsPipe.decorators = [
        { type: i0.Pipe, args: [{
                    name: 'shopsPipe'
                },] }
    ];
    ShopsPipe.ctorParameters = function () { return [
        { type: UserService }
    ]; };

    var OnFetchComponent = /** @class */ (function () {
        function OnFetchComponent() {
            this.refreshCallback = new i0.EventEmitter();
        }
        OnFetchComponent.prototype.ngOnInit = function () {
        };
        OnFetchComponent.prototype.callBack = function () {
            this.refreshCallback.emit();
        };
        return OnFetchComponent;
    }());
    OnFetchComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-on-fetch',
                    template: "\n    <div *ngIf=\"!isLoading\" class=\"no-data-wrapper\">\n      <img src=\"/src/assets/img/empty.svg\" width=\"200px\" alt=\"\">\n      <h4>No Data</h4>\n      <button (click)=\"callBack()\" matTooltip=\"Click to refresh\" mat-raised-button mat-icon-button>\n        <mat-icon color=\"accent\">\n          refresh\n        </mat-icon>\n      </button>\n    </div>\n\n    <div *ngIf=\"isLoading\" class=\"no-data-wrapper\">\n      <img src=\"/src/assets/img/data.svg\" width=\"200px\" alt=\"\">\n      <h4>Fetching data...</h4>\n      <mat-progress-spinner color=\"accent\" mode=\"indeterminate\" [diameter]=\"30\"></mat-progress-spinner>\n    </div>\n  ",
                    styles: [".no-data-wrapper{align-items:center;display:flex;flex-direction:column;justify-content:center;margin-top:48px}.no-data-wrapper>h4{padding:8px}"]
                },] }
    ];
    OnFetchComponent.ctorParameters = function () { return []; };
    OnFetchComponent.propDecorators = {
        refreshCallback: [{ type: i0.Output }],
        isLoading: [{ type: i0.Input }]
    };

    var SearchInputComponent = /** @class */ (function () {
        function SearchInputComponent() {
            this.showSearch = false;
            this.searchProgressFlag = false;
            this.searchPlaceholder = 'Enter something...';
        }
        SearchInputComponent.prototype.ngOnInit = function () {
        };
        return SearchInputComponent;
    }());
    SearchInputComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-search-input',
                    template: "\n    <div style=\"width: 100%\">\n      <input [formControl]=\"searchInputControl\"\n             *ngIf=\"showSearch\"\n             class=\"toolbar-input\" type=\"text\" placeholder=\"{{searchPlaceholder}}\">\n      <mat-progress-spinner style=\"display: inline-block; margin-left: -30px\"\n                            mode=\"indeterminate\" diameter=\"25\"\n                            color=\"primary\"\n                            *ngIf=\"showSearch && searchProgressFlag\"\n                            matSuffix>\n        <!--      <mat-icon>search</mat-icon>-->\n      </mat-progress-spinner>\n    </div>\n  ",
                    styles: [".toolbar-input{background:hsla(0,0%,96.5%,.76);border:none;border-radius:4px;box-shadow:none;color:#000;outline:none;padding:8px;width:100%}.toolbar-input::-moz-placeholder{color:#535353}.toolbar-input:-ms-input-placeholder{color:#535353}.toolbar-input::placeholder{color:#535353}.toolbar-input:focus{outline:none}"]
                },] }
    ];
    SearchInputComponent.ctorParameters = function () { return []; };
    SearchInputComponent.propDecorators = {
        searchInputControl: [{ type: i0.Input }],
        showSearch: [{ type: i0.Input }],
        searchProgressFlag: [{ type: i0.Input }],
        searchPlaceholder: [{ type: i0.Input }]
    };

    var BottomBarComponent = /** @class */ (function () {
        function BottomBarComponent() {
        }
        BottomBarComponent.prototype.ngOnInit = function () {
        };
        BottomBarComponent.prototype.activeNav = function (route) {
            var url = new URL(location.href);
            return url.pathname.startsWith('/' + route);
        };
        return BottomBarComponent;
    }());
    BottomBarComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-bottom-bar',
                    template: "\n    <mat-toolbar class=\"fixed-bottom bottom-smartstock-bar\">\n\n      <div class=\"flex-grow-1 d-flex justify-content-center align-items-center flex-column\">\n        <button mat-icon-button *ngIf=\"activeNav('dashboard')\" mat-flat-button routerLink=\"/dashboard\">\n          <mat-icon color=\"{{activeNav('dashboard')?'primary':''}}\">dashboard</mat-icon>\n        </button>\n        <button *ngIf=\"!activeNav('dashboard')\" mat-icon-button routerLink=\"/dashboard\">\n          <mat-icon>dashboard</mat-icon>\n        </button>\n        <!--    <span *ngIf=\"activeNav('dashboard')\">Dashboard</span>-->\n      </div>\n\n      <div class=\"flex-grow-1 d-flex justify-content-center align-items-center flex-column\">\n        <button mat-icon-button *ngIf=\"activeNav('sale')\" mat-flat-button routerLink=\"/sale\">\n          <mat-icon color=\"{{activeNav('sale')?'primary':''}}\">shop</mat-icon>\n        </button>\n        <button *ngIf=\"!activeNav('sale')\" mat-icon-button routerLink=\"/sale\">\n          <mat-icon>shop</mat-icon>\n        </button>\n        <!--    <span *ngIf=\"activeNav('sale')\">Sale</span>-->\n      </div>\n\n      <div class=\"flex-grow-1 d-flex justify-content-center align-items-center flex-column\">\n        <button mat-icon-button *ngIf=\"activeNav('purchase')\" mat-flat-button routerLink=\"/purchase\">\n          <mat-icon color=\"{{activeNav('purchase')?'primary':''}}\">receipt</mat-icon>\n        </button>\n        <button *ngIf=\"!activeNav('purchase')\" mat-icon-button routerLink=\"/purchase\">\n          <mat-icon>receipt</mat-icon>\n        </button>\n        <!--    <span *ngIf=\"activeNav('purchase')\">Purchase</span>-->\n      </div>\n\n      <div class=\"flex-grow-1 d-flex justify-content-center align-items-center flex-column\">\n        <button mat-icon-button *ngIf=\"activeNav('stock')\" mat-flat-button routerLink=\"/stock\">\n          <mat-icon color=\"{{activeNav('stock')?'primary':''}}\">store</mat-icon>\n        </button>\n        <button *ngIf=\"!activeNav('stock')\" mat-icon-button routerLink=\"/stock\">\n          <mat-icon>store</mat-icon>\n        </button>\n        <!--    <span *ngIf=\"activeNav('stock')\">StockModel</span>-->\n      </div>\n\n      <div class=\"flex-grow-1 d-flex justify-content-center align-items-center flex-column\">\n        <button mat-icon-button *ngIf=\"activeNav('settings')\" mat-flat-button routerLink=\"/account\">\n          <mat-icon color=\"{{activeNav('settings')?'primary':''}}\">settings</mat-icon>\n        </button>\n        <button *ngIf=\"!activeNav('settings')\" mat-icon-button routerLink=\"/account\">\n          <mat-icon>settings</mat-icon>\n        </button>\n        <!--    <span *ngIf=\"activeNav('settings')\">Settings</span>-->\n      </div>\n\n    </mat-toolbar>\n  ",
                    styles: ["span{font-size:11px;font-weight:lighter}.bottom-smartstock-bar{background:#fff;box-shadow:0 -1px 7px rgba(0,0,0,.35)}"]
                },] }
    ];
    BottomBarComponent.ctorParameters = function () { return []; };

    var DataNotReadyComponent = /** @class */ (function () {
        function DataNotReadyComponent() {
            this.width = 200;
            this.height = 200;
            this.isLoading = false;
        }
        DataNotReadyComponent.prototype.ngOnInit = function () {
        };
        return DataNotReadyComponent;
    }());
    DataNotReadyComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-data-not-ready',
                    template: "\n    <div *ngIf=\"!isLoading\" style=\"margin-top: 10px; margin-bottom: 10px\"\n         class=\"d-flex flex-column justify-content-center align-items-center\">\n      <img alt=\"no data\" src=\"../../../../assets/img/data.svg\" width=\"{{width}}\" height=\"{{height}}\"/>\n      <span>Data is not ready yet</span>\n    </div>\n    <div *ngIf=\"isLoading\" style=\"width: 100%; height: 100%; display: flex; justify-content: center; align-items: center\">\n      <mat-progress-spinner mode=\"indeterminate\" [diameter]=\"30\"></mat-progress-spinner>\n    </div>\n  "
                },] }
    ];
    DataNotReadyComponent.ctorParameters = function () { return []; };
    DataNotReadyComponent.propDecorators = {
        width: [{ type: i0.Input }],
        height: [{ type: i0.Input }],
        isLoading: [{ type: i0.Input }]
    };

    var UploadFilesComponent = /** @class */ (function () {
        function UploadFilesComponent() {
            this.files = [];
            this.uploadFileFormControl = new forms.FormControl([], [forms.Validators.nullValidator, forms.Validators.required]);
        }
        UploadFilesComponent.prototype.ngOnInit = function () {
        };
        UploadFilesComponent.prototype.removeFile = function ($event, i) {
            $event.preventDefault();
            this.files.splice(i, 1);
        };
        UploadFilesComponent.prototype.uploadFiles = function ($event, uploadFile) {
            return __awaiter(this, void 0, void 0, function () {
                var files, file_1;
                return __generator(this, function (_a) {
                    files = $event.target['files'];
                    if (files.item(0)) {
                        file_1 = files.item(0);
                        if (this.files.length === 0) {
                            this.files.push({
                                name: file_1.name,
                                type: file_1.type,
                                url: file_1,
                            });
                        }
                        else {
                            this.files = this.files.filter(function (value) { return file_1.name !== value.name; });
                            this.files.push({
                                name: file_1.name,
                                type: file_1.type,
                                url: file_1
                            });
                        }
                        this.uploadFileFormControl.setValue(this.files);
                        uploadFile.value = '';
                    }
                    return [2 /*return*/];
                });
            });
        };
        return UploadFilesComponent;
    }());
    UploadFilesComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-upload-files',
                    template: "\n    <div style=\"display: flex; flex-direction: row; flex-wrap: wrap; align-items: center; padding: 8px\">\n      <div *ngFor=\"let file of files; let i =index\"\n           style=\"display: flex; flex-direction: row; flex-wrap: nowrap; align-items: center\">\n        <mat-card style=\"height: 50px; margin: 5px; display: flex; flex-direction: row; align-items: center\">\n          <span style=\"max-width: 200px; overflow: hidden; white-space: nowrap; text-overflow: ellipsis\">\n            {{file.name}}\n          </span>\n          <span style=\"width: 10px\"></span>\n          <button mat-icon-button style=\"display: inline-block\" (click)=\"removeFile($event, i)\">\n            <mat-icon color=\"warn\">delete</mat-icon>\n          </button>\n        </mat-card>\n      </div>\n      <mat-card (click)=\"uploadFile.click()\" matRipple style=\"width: 120px; height: 50px; margin: 5px\">\n        <mat-icon>attachment</mat-icon>\n        <span>Add File</span>\n      </mat-card>\n    </div>\n\n    <input style=\"display: none\" #uploadFile [multiple]=\"false\" type=\"file\" (change)=\"uploadFiles($event,uploadFile)\">\n  "
                },] }
    ];
    UploadFilesComponent.ctorParameters = function () { return []; };
    UploadFilesComponent.propDecorators = {
        files: [{ type: i0.Input }],
        uploadFileFormControl: [{ type: i0.Input }]
    };

    var DashCardComponent = /** @class */ (function () {
        function DashCardComponent(router) {
            this.router = router;
            this.title = '';
            this.height = 300;
            this.description = '';
        }
        DashCardComponent.prototype.ngOnInit = function () {
        };
        DashCardComponent.prototype.goTo = function () {
            if (this.reportLink) {
                this.router.navigateByUrl(this.reportLink).catch(console.log);
            }
        };
        return DashCardComponent;
    }());
    DashCardComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-dash-card',
                    template: "\n    <mat-card class=\"d-flex flex-column\" [ngStyle]=\"{height:height+'px'}\">\n      <mat-card-header style=\"margin: 0\">\n        <span>{{title}}</span>\n        <span style=\"margin-left: 8px\"></span>\n        <mat-icon matTooltip=\"{{description}}\">info_outline</mat-icon>\n      </mat-card-header>\n      <mat-card-content class=\"flex-grow-1\">\n        <ng-container [ngTemplateOutlet]=\"content\"></ng-container>\n      </mat-card-content>\n      <hr *ngIf=\"reportLink\" style=\"width: 100%; margin: 0; padding: 0\">\n      <mat-card-actions *ngIf=\"reportLink\" class=\"d-flex flex-row\">\n        <span style=\"flex: 1 1 auto\"></span>\n        <button (click)=\"goTo()\"color=\"primary\" mat-button>\n          EXPLORE REPORT\n          <mat-icon matSuffix>arrow_forward</mat-icon>\n        </button>\n      </mat-card-actions>\n    </mat-card>\n  "
                },] }
    ];
    DashCardComponent.ctorParameters = function () { return [
        { type: router.Router }
    ]; };
    DashCardComponent.propDecorators = {
        title: [{ type: i0.Input }],
        height: [{ type: i0.Input }],
        description: [{ type: i0.Input }],
        reportLink: [{ type: i0.Input }],
        content: [{ type: i0.Input }]
    };

    var UploadFileProgressComponent = /** @class */ (function () {
        function UploadFileProgressComponent() {
            this.name = '';
            this.onUploadFlag = false;
            this.uploadPercentage = 0;
        }
        UploadFileProgressComponent.prototype.ngOnInit = function () {
        };
        return UploadFileProgressComponent;
    }());
    UploadFileProgressComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-upload-file-progress',
                    template: "\n    <div *ngIf=\"onUploadFlag\">\n      <mat-progress-bar [bufferValue]=\"100\" mode=\"determinate\" [value]=\"uploadPercentage\"></mat-progress-bar>\n      <span>{{name}} {{uploadPercentage}}%</span>\n    </div>\n  "
                },] }
    ];
    UploadFileProgressComponent.ctorParameters = function () { return []; };
    UploadFileProgressComponent.propDecorators = {
        name: [{ type: i0.Input }],
        onUploadFlag: [{ type: i0.Input }],
        uploadPercentage: [{ type: i0.Input }]
    };

    var DialogDeleteComponent = /** @class */ (function () {
        function DialogDeleteComponent(dialogRef, data) {
            this.dialogRef = dialogRef;
            this.data = data;
        }
        DialogDeleteComponent.prototype.delete = function (stock) {
            this.dialogRef.close(stock);
        };
        DialogDeleteComponent.prototype.cancel = function () {
            this.dialogRef.close('no');
        };
        return DialogDeleteComponent;
    }());
    DialogDeleteComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'smartstock-lisb-dialog-delete',
                    template: "\n    <div class=\"container\">\n      <div class=\"row\">\n        <div class=\"col-12\">\n          <mat-panel-title class=\"text-center\">\n            Your about to delete :  <b>{{' '+ data.product}}</b>\n          </mat-panel-title>\n        </div>\n      </div>\n      <div class=\"d-flex justify-content-center\">\n        <div class=\"align-self-center\" style=\"margin: 8px\">\n          <button color=\"primary\"  mat-button (click)=\"delete(data)\">Delete</button>\n        </div>\n        <div class=\"alert-secondary\" style=\"margin: 8px\">\n          <button color=\"primary\" mat-button (click)=\"cancel()\">Cancel</button>\n        </div>\n      </div>\n    </div>\n  "
                },] }
    ];
    DialogDeleteComponent.ctorParameters = function () { return [
        { type: i3.MatDialogRef },
        { type: undefined, decorators: [{ type: i0.Inject, args: [i3.MAT_DIALOG_DATA,] }] }
    ]; };

    var LibModule = /** @class */ (function () {
        function LibModule() {
        }
        LibModule.start = function (config) {
            if (config === void 0) { config = { version: '' }; }
            ConfigsService.versionName = config.version;
        };
        return LibModule;
    }());
    LibModule.decorators = [
        { type: i0.NgModule, args: [{
                    imports: [
                        common.CommonModule,
                        progressSpinner.MatProgressSpinnerModule,
                        card.MatCardModule,
                        expansion.MatExpansionModule,
                        divider.MatDividerModule,
                        list.MatListModule,
                        progressBar.MatProgressBarModule,
                        toolbar.MatToolbarModule,
                        icon.MatIconModule,
                        menu.MatMenuModule,
                        button.MatButtonModule,
                        router.RouterModule,
                        formField.MatFormFieldModule,
                        forms.ReactiveFormsModule,
                        i3.MatDialogModule,
                        select.MatSelectModule,
                        badge.MatBadgeModule,
                        tooltip.MatTooltipModule,
                        core.MatRippleModule
                    ],
                    exports: [
                        DrawerComponent,
                        ToolbarComponent,
                        ShopsPipe,
                        OnFetchComponent,
                        BottomBarComponent,
                        DataNotReadyComponent,
                        UploadFilesComponent,
                        DashCardComponent,
                        UploadFileProgressComponent
                    ],
                    declarations: [
                        DrawerComponent,
                        ToolbarComponent,
                        NoStockDialogComponent,
                        ShopsPipe,
                        OnFetchComponent,
                        SearchInputComponent,
                        BottomBarComponent,
                        DataNotReadyComponent,
                        UploadFilesComponent,
                        DashCardComponent,
                        DialogDeleteComponent,
                        UploadFileProgressComponent,
                        VerifyEMailDialogComponent
                    ]
                },] }
    ];

    var FileLibraryService = /** @class */ (function () {
        function FileLibraryService(storageService) {
            this.storageService = storageService;
        }
        FileLibraryService.prototype.saveFile = function (file, progress) {
            return __awaiter(this, void 0, void 0, function () {
                var activeShop;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            if (!(file && file instanceof File)) return [3 /*break*/, 2];
                            return [4 /*yield*/, this.storageService.getActiveShop()];
                        case 1:
                            activeShop = _a.sent();
                            return [2 /*return*/, bfastjs.BFast.storage(activeShop === null || activeShop === void 0 ? void 0 : activeShop.projectId).save(file, function (data) {
                                    progress((Number(data.loaded) / Number(data.total) * 100));
                                })];
                        case 2:
                            if (file && typeof file === 'string' && file.startsWith('http')) {
                                return [2 /*return*/, file];
                            }
                            else {
                                return [2 /*return*/, null];
                            }
                            _a.label = 3;
                        case 3: return [2 /*return*/];
                    }
                });
            });
        };
        return FileLibraryService;
    }());
    FileLibraryService.ɵprov = i0.ɵɵdefineInjectable({ factory: function FileLibraryService_Factory() { return new FileLibraryService(i0.ɵɵinject(StorageService)); }, token: FileLibraryService, providedIn: "any" });
    FileLibraryService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'any'
                },] }
    ];
    FileLibraryService.ctorParameters = function () { return [
        { type: StorageService }
    ]; };

    var MessageService = /** @class */ (function () {
        function MessageService(matSnackBar) {
            this.matSnackBar = matSnackBar;
        }
        MessageService.prototype.showMobileInfoMessage = function (message, duration, position) {
            if (duration === void 0) { duration = 3000; }
            if (position === void 0) { position = 'top'; }
            this.matSnackBar.open(message, 'Ok', {
                horizontalPosition: 'center',
                verticalPosition: position,
                duration: duration
            });
        };
        return MessageService;
    }());
    MessageService.ɵprov = i0.ɵɵdefineInjectable({ factory: function MessageService_Factory() { return new MessageService(i0.ɵɵinject(i1$1.MatSnackBar)); }, token: MessageService, providedIn: "root" });
    MessageService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    MessageService.ctorParameters = function () { return [
        { type: i1$1.MatSnackBar }
    ]; };

    var toSqlDate = function (date) {
        var year = date.getFullYear();
        var month = (date.getMonth() + 1).toString(10);
        var day = (date.getDate()).toString(10);
        if (month.length === 1) {
            month = '0' + month;
        }
        if (day.length === 1) {
            day = '0' + day;
        }
        return year + '-' + month + '-' + day;
    };

    var DeviceInfoUtil = /** @class */ (function () {
        function DeviceInfoUtil() {
        }
        DeviceInfoUtil.prototype.getWidth = function () {
            return window.innerWidth;
        };
        DeviceInfoUtil.prototype.enoughWidth = function () {
            var width = this.getWidth();
            return width > 1000;
        };
        DeviceInfoUtil.prototype.openDrawer = function (sidenav) {
            sidenav.open().catch();
        };
        return DeviceInfoUtil;
    }());

    /*
     * Public API Surface of libs
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.DeviceInfoUtil = DeviceInfoUtil;
    exports.EventService = EventService;
    exports.FileLibraryService = FileLibraryService;
    exports.LibModule = LibModule;
    exports.LogService = LogService;
    exports.MessageService = MessageService;
    exports.NoStockDialogComponent = NoStockDialogComponent;
    exports.SecurityUtil = SecurityUtil;
    exports.SettingsService = SettingsService;
    exports.SsmEvents = SsmEvents;
    exports.StorageService = StorageService;
    exports.UserService = UserService;
    exports.toSqlDate = toSqlDate;
    exports.ɵa = DrawerComponent;
    exports.ɵb = ToolbarComponent;
    exports.ɵc = ShopsPipe;
    exports.ɵd = OnFetchComponent;
    exports.ɵe = BottomBarComponent;
    exports.ɵf = DataNotReadyComponent;
    exports.ɵg = UploadFilesComponent;
    exports.ɵh = DashCardComponent;
    exports.ɵi = UploadFileProgressComponent;
    exports.ɵj = SearchInputComponent;
    exports.ɵk = DialogDeleteComponent;
    exports.ɵm = VerifyEMailDialogComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
